### PR TITLE
ROX-20320: remove Clair v2 integration

### PR DESCRIPTION
**Pull request by RTann**
## Description

Clair v2 integration has been deprecated for several releases, so it is time to finally remove it.

## Checklist
- [x] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- [x] Evaluated and added CHANGELOG entry if required
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

## Testing Performed

Check the UI, and you'll see it is gone.

### Upgrade tests

```
# Deploy 4.2.1. Be sure you have STORAGE=pvc -
$ MAIN_IMAGE_TAG=4.2.1 ./deploy/k8s/deploy-local.sh
# Deploy Clair v2 -
$ POD_SECURITY_POLICIES=false CLAIR_DB_PASSWORD=password ./scripts/ci/clair/deploy.sh qa-clair
# In the Image Integration UI, create a CoreOS Clair integration pointing to http://clairsvc.qa-clair:6060
# Create API token (bottom of Integrations page) for the Admin role
$ export ROX_API_TOKEN=<long token generated in previous step>
$ ./bin/<your arch here>/roxctl -e "localhost:8000" central backup
$ ./bin/<your arch here>/roxctl -e "localhost:8000" central db generate k8s pvc --output-dir tmp --central-db-image quay.io/stackrox-io/central-db
$ ./tmp/scripts/setup.sh
$ ./tmp/deploy-central-db.sh
$ kubectl -n stackrox set image deploy/central central=quay.io/stackrox-io/main:4.2.x-539-g9e4ae80ee6
# Verify Central is running -
$ kubectl -n stackrox logs -f deploy/central
```

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by pedrottimark — APPROVED** [review]
Frontend changes make sense.

**Review by gavin-stackrox — APPROVED** [review]
Groovy changes look groovy

**Review by dcaravel — COMMENTED** [review]
Are we confident that removing Clair from the storage protos will not cause issues if `central-db` has integrations still referencing clair?  Is a migration needed to cleanup any lingering integrations?

**Comment by RTann** [comment]
> Are we confident that removing Clair from the storage protos will not cause issues if `central-db` has integrations still referencing clair? Is a migration needed to cleanup any lingering integrations from DB?

good question. I'll request a review from someone on Core Workflows

**Review by connorgorman — COMMENTED** [review]
Please ensure that if one is integrated that it doesn't cause a crash when it doesn't exist and instead is silently ignored

File: central/imageintegration/datastore/singleton.go [hunk 1]
```diff
@@ -24,8 +24,19 @@ func initializeIntegrations(storage store.Store) {
 	ctx := sac.WithGlobalAccessScopeChecker(context.Background(), sac.AllowAllAccessScopeChecker())
 	iis, err := storage.GetAll(ctx)
 	utils.CrashOnError(err)
+
+	// If we are starting from scratch, it means there are no preexisting integrations.
+	fromScratch := len(iis) == 0
+	// Ensure there are no preexisting image integrations which are no longer supported.
+	if !fromScratch {
+		for _, ii := range iis {
+			if ii.GetIntegrationConfig() == nil {
```
Comment (RTann) [inline central/imageintegration/datastore/singleton.go:33]: Need to test this in an upgrade scenario

**Pull request closed without merging**